json-schema-provider/src/main/java/io/confluent/kafka/schemaregistry/json/JsonSchemaUtils.java

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.kafka.schemaregistry.json;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.JsonNodeFactory;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.kjetland.jackson.jsonSchema.JsonSchemaConfig;
+import com.kjetland.jackson.jsonSchema.JsonSchemaGenerator;
+
+import java.io.IOException;
+import java.io.StringWriter;
+import java.nio.charset.StandardCharsets;
+
+import io.confluent.kafka.schemaregistry.ParsedSchema;
+import io.confluent.kafka.schemaregistry.json.jackson.Jackson;
+
+public class JsonSchemaUtils {
+
+  private static final Object NONE_MARKER = new Object();
+
+  private static final ObjectMapper jsonMapper = Jackson.newObjectMapper();
+
+  static final String ENVELOPE_SCHEMA_FIELD_NAME = "schema";
+  static final String ENVELOPE_PAYLOAD_FIELD_NAME = "payload";
+
+  public static ObjectNode envelope(JsonNode schema, JsonNode payload) {
+    ObjectNode result = JsonNodeFactory.instance.objectNode();
+    result.set(ENVELOPE_SCHEMA_FIELD_NAME, schema);
+    result.set(ENVELOPE_PAYLOAD_FIELD_NAME, payload);
+    return result;
+  }
+
+  public static JsonSchema copyOf(JsonSchema schema) {
+    return new JsonSchema(schema.canonicalString(),
+        schema.references(),
+        schema.resolvedReferences(),
+        schema.version()
+    );
+  }
+
+  public static JsonSchema getSchema(Object object) {
+    if (object == null) {
+      return null;
+    }
+    if (object instanceof JsonNode) {
+      JsonNode jsonValue = (JsonNode) object;
+      if (jsonValue.isObject() && jsonValue.has(ENVELOPE_SCHEMA_FIELD_NAME)) {
+        return new JsonSchema(jsonValue.get(ENVELOPE_SCHEMA_FIELD_NAME).toString());
+      }
+    }
+    JsonSchemaConfig config = JsonSchemaConfig.nullableJsonSchemaDraft4();  // allow nulls
```

Comment:
Are there plans to support higher drafts?

Reply:
This is just a config on the underlying library to allow nullable types for draft 4 or later.

Reply:
> draft 4 or later

IIRC, draft 6 & 7 are not backwards compatible with 4, so not sure this would support "or later"

Reply:
The config is a bit misnamed :  `Under the hood nullableJsonSchemaDraft4 toggles the useOneOfForOption and useOneOfForNullables properties on JsonSchemaConfig.` from https://github.com/mbknor/mbknor-jackson-jsonSchema/